### PR TITLE
Don't use deprecated querystring package

### PR DESCRIPTION
**Pull request by mischnic**
`URLSearchParams` is supported in Node >=10

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.22s | -152.00ms 🚀 |
| Cached | 320.00ms | -10.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 676.00ms | +591.00ms ⚠️ |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 677.00ms | +591.00ms ⚠️ |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 678.00ms | +592.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 11.04s | -3.00ms |
| Cached | 474.00ms | -3.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.20m | -326.00ms |
| Cached | 1.81s | +103.00ms ⚠️ |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/editorView.229c5c5f.js | 594.84kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/popup.520c7863.js | 209.67kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/EmojiPickerComponent.4259176d.js | 146.67kb | +0.00b | 34.08s | +11.66s ⚠️ |
| dist/Toolbar.7c4d7cd4.js | 107.23kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/Modal.77365192.js | 45.33kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/esm.304eb046.js | 33.15kb | +0.00b | 34.09s | +11.67s ⚠️ |
| dist/DatePicker.05d89a3e.js | 22.98kb | +0.00b | 34.08s | +11.66s ⚠️ |
| dist/ui.6d0d68f8.js | 14.94kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/smartMediaEditor.8cebfbfd.js | 13.25kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/dropzone.0d183115.js | 12.15kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/EmojiPickerComponent.55e0a64c.js | 3.73kb | +0.00b | 41.14s | -13.99s 🚀 |
| dist/dropzone.cf53fd41.js | 3.29kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/clipboard.969d5ad7.js | 2.93kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/ResourcedEmojiComponent.78f6ddfc.js | 2.12kb | +0.00b | 41.14s | -13.99s 🚀 |
| dist/feedback.e18b45f0.js | 1.77kb | +0.00b | 34.08s | +11.66s ⚠️ |
| dist/browser.09200cd5.js | 1.69kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/workerHasher.567cfc57.js | 1.63kb | +0.00b | 34.09s | +11.67s ⚠️ |
| dist/heading6.b3b946d1.js | 1.51kb | +0.00b | 34.08s | +11.66s ⚠️ |
| dist/heading5.e617db66.js | 1.38kb | +0.00b | 34.08s | +11.66s ⚠️ |
| dist/expand.1f17de7f.js | 1.29kb | +0.00b | 34.08s | +11.66s ⚠️ |
| dist/media-card-analytics-error-boundary.75fac461.js | 1.12kb | +0.00b | 41.15s | -13.98s 🚀 |
| dist/media-picker-analytics-error-boundary.cfe02cc6.js | 966.00b | +0.00b | 41.15s | -13.98s 🚀 |
| dist/simpleHasher.2a6e12b6.js | 643.00b | +0.00b | 41.14s | -13.99s 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/editorView.f2107629.js | 594.84kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/popup.520c7863.js | 209.67kb | +0.00b | 55.30s | +13.65s ⚠️ |
| dist/Toolbar.7c4d7cd4.js | 107.23kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/card.d1c4b983.js | 62.67kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/Modal.77365192.js | 45.33kb | +0.00b | 55.30s | +13.65s ⚠️ |
| dist/ui.6d0d68f8.js | 14.94kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/smartMediaEditor.8cebfbfd.js | 13.25kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/dropzone.0d183115.js | 12.15kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/card.45f850d6.js | 5.77kb | +0.00b | 55.29s | +13.64s ⚠️ |
| dist/EmojiPickerComponent.55e0a64c.js | 3.73kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/dropzone.cf53fd41.js | 3.29kb | +0.00b | 55.30s | +13.65s ⚠️ |
| dist/Modal.7f4c60b3.js | 3.15kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/clipboard.969d5ad7.js | 2.93kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/ResourcedEmojiComponent.78f6ddfc.js | 2.12kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/browser.09200cd5.js | 1.69kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/media-card-analytics-error-boundary.75fac461.js | 1.12kb | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/media-picker-analytics-error-boundary.cfe02cc6.js | 966.00b | +0.00b | 55.30s | +13.65s ⚠️ |
| dist/media-card-analytics-error-boundary.b1a780f3.js | 962.00b | +0.00b | 55.30s | +13.64s ⚠️ |
| dist/simpleHasher.2a6e12b6.js | 643.00b | +0.00b | 55.30s | +13.64s ⚠️ |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.57s | -30.00ms |
| Cached | 432.00ms | +17.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/311759826705187392)**

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.99s | -184.00ms 🚀 |
| Cached | 332.00ms | +17.00ms ⚠️ |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/index.5ba75694.css | 77.00b | +0.00b | 1.14s | -136.00ms 🚀 |
| dist/modern/index.14235cd8.css | 77.00b | +0.00b | 1.14s | -137.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 84.00ms | -542.00ms 🚀 |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 85.00ms | -541.00ms 🚀 |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 84.00ms | -543.00ms 🚀 |
| dist/legacy/index.9c1188a7.js | 1.54kb | +0.00b | 854.00ms | -45.00ms 🚀 |
| dist/legacy/index.9106df33.js | 1.15kb | +0.00b | 854.00ms | -44.00ms 🚀 |
| dist/modern/index.0345151f.js | 1.08kb | +0.00b | 853.00ms | -45.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 10.88s | +903.00ms ⚠️ |
| Cached | 468.00ms | +21.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 486.94kb | +0.00b | 6.06s | +661.00ms ⚠️ |
| dist/PermalinkedComment.e846300d.js | 4.18kb | +0.00b | 6.06s | +660.00ms ⚠️ |
| dist/UserProfile.a136423a.js | 1.56kb | +0.00b | 6.06s | +660.00ms ⚠️ |
| dist/NotFound.a7537975.js | 412.00b | +0.00b | 6.06s | +661.00ms ⚠️ |
| dist/logo.c5bb83f1.png | 246.00b | +0.00b | 6.00s | +658.00ms ⚠️ |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.17m | +2.66s |
| Cached | 1.62s | +1.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.514dca99.js | 1.77mb | -10.00b 🚀 | 20.86s | +1.36s ⚠️ |
| dist/index.2065c44c.js | 694.58kb | +4.00b ⚠️ | 53.26s | +1.26s |
| dist/component.f1b41b58.js | 37.33kb | +0.00b | 20.86s | +1.36s ⚠️ |
| dist/component.a0f9f96f.js | 24.93kb | +0.00b | 20.86s | +1.36s ⚠️ |
| dist/component.2de9ac3a.js | 6.81kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.7b42ca8a.js | 2.33kb | +0.00b | 20.86s | +1.36s ⚠️ |
| dist/date.c0482a02.js | 1.85kb | +0.00b | 20.85s | +1.36s ⚠️ |
| dist/images.c21e6480.js | 1.79kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.f3960d82.js | 1.72kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.3aec39cf.js | 1.66kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/list-number.e0420e0c.js | 1.58kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/status.f223f976.js | 1.58kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/code.cd95a955.js | 1.50kb | +0.00b | 20.85s | +1.36s ⚠️ |
| dist/heading3.33ac6e42.js | 1.48kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/link.7487dc5e.js | 1.42kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.332fbe28.js | 1.37kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/emoji.03d3d99b.js | 1.34kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.79a0250b.js | 1.33kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.a877725f.js | 1.33kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.0eecfc6b.js | 1.30kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/heading2.d374048d.js | 1.30kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.01994bf0.js | 1.30kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.00ba62db.js | 1.27kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/heading4.62b9719b.js | 1.25kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.8e758d30.js | 1.22kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.0555a5c0.js | 1.19kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.b114a692.js | 1.18kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/mention.37efe6c4.js | 1.18kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.e21a2218.js | 1.18kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/layout.01a386bd.js | 1.16kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/heading1.a1828016.js | 1.15kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/divider.601227e8.js | 1.14kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/quote.bb9e77b1.js | 1.14kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.922eddd1.js | 1.13kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.e92739d9.js | 1.13kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.92a42544.js | 1.12kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.e5c3bc4c.js | 1.12kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/16.0358f7c7.js | 1.12kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/action.9e337641.js | 1.12kb | +0.00b | 20.86s | +1.36s ⚠️ |
| dist/16.455c97fa.js | 1.12kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/component.edb11b95.js | 1.11kb | +0.00b | 20.86s | +1.36s ⚠️ |
| dist/decision.9b91a34d.js | 1.10kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/panel-warning.2ec3e2eb.js | 1.10kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.b221eda2.js | 1.08kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/list.13abafe4.js | 1.07kb | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/16.cc9a820b.js | 1.04kb | +0.00b | 20.58s | +1.30s ⚠️ |
| dist/panel-error.9e60042f.js | 1024.00b | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/panel.d9a0dff6.js | 1015.00b | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/table.10221eb5.js | 1007.00b | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/panel-success.6b7c55df.js | 963.00b | +0.00b | 22.53s | +1.53s ⚠️ |
| dist/panel-note.1f570859.js | 959.00b | +0.00b | 22.53s | +1.53s ⚠️ |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.24s | -88.00ms |
| Cached | 409.00ms | +35.00ms ⚠️ |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/308269432384258626)**

File: packages/utils/node-resolver-core/src/NodeResolver.js [hunk 1]
```diff
@@ -526,7 +525,9 @@ export default class NodeResolver {
 
       return {
         filePath,
-        query: url.search ? parseQueryString(url.search.slice(1)) : undefined,
+        query: url.search
+          ? Object.fromEntries(new URLSearchParams(url.search.slice(1)))
```
Comment (devongovett) [inline packages/utils/node-resolver-core/src/NodeResolver.js]: Is it a problem that this doesn't handle duplicates the same way?

```
> require('querystring').parse('a=b&a=c')
[Object: null prototype] { a: [ 'b', 'c' ] }

> Object.fromEntries(new URLSearchParams('a=b&a=c'))
{ a: 'c' }
```

Comment (mischnic) [reply packages/utils/node-resolver-core/src/NodeResolver.js]: I forgot about that. Then we should also change
https://github.com/parcel-bundler/parcel/blob/461de1149a0312d4c541d8e5fca977a7f3c58f7e/packages/core/types/index.js#L28
to be `{ [string]: string | Array<string> }` as well

File: packages/transformers/image/src/ImageTransformer.js [hunk 2]
```diff
@@ -21,12 +21,22 @@ export default (new Transformer({
...
-    let width = asset.query.width ? parseInt(asset.query.width, 10) : null;
-    let height = asset.query.height ? parseInt(asset.query.height, 10) : null;
-    let quality = asset.query.quality
-      ? parseInt(asset.query.quality, 10)
-      : undefined;
-    let format = asset.query.as ? asset.query.as.toLowerCase().trim() : null;
+    let width =
+      typeof asset.query.width === 'string'
+        ? parseInt(asset.query.width, 10)
+        : null;
```
Comment (devongovett) [inline packages/transformers/image/src/ImageTransformer.js]: ugh that's annoying... should we even support duplicate query params?

Or maybe `asset.query` should be a `URLSearchParams` object?

**Review by devongovett — COMMENTED** [review]
Isn't the querystring package built into node? https://nodejs.org/api/querystring.html

**Comment by mischnic** [comment]
> Isn't the querystring package built into node? nodejs.org/api/querystring.html

It says:

> Stability: 3 - Legacy
>
> The feature is no longer recommended for use. While it likely will not be removed, and is still covered by semantic-versioning guarantees, use of the feature should be avoided.

**Review by devongovett — APPROVED** [review]

**Pull request merged**